montepy/surfaces/surface.py

```diff
@@ -35,68 +45,154 @@ def __init__(
-                    self._old_periodic_surface = val
-                else:
-                    self._old_transform_number = val
-            self._surface_type = self._tree["surface_type"]
-            # parse surface mnemonic
-            try:
-                # enforce enums
-                self._surface_type._convert_to_enum(SurfaceType, switch_to_upper=True)
-            # this should never be reached due to SLY rules.
-            # still if it is somehow reached this error is more helpful to the user.
-            except ValueError:  # pragma: no cover
-                raise MalformedInputError(
-                    input,
-                    f"{self._surface_type.value} could not be parsed as a surface type mnemonic.",
-                )
-            # parse the parameters
-            for entry in self._tree["data"]:
-                self._surface_constants.append(entry)
+        except AssertionError:
+            raise MalformedInputError(
+                input,
+                f"{self._number.value} could not be parsed as a valid surface number.",
+            )
+        if self._tree["pointer"].value is not None:
+            val = self._tree["pointer"]
+            val.is_negatable_identifier = True
+            if val.is_negative:
+                self._old_periodic_surface = val
+            else:
+                self._old_transform_number = val
+        self._surface_type = self._tree["surface_type"]
+        # parse surface mnemonic
+        try:
+            # enforce enums
+            self._surface_type._convert_to_enum(
+                SurfaceType, allow_none=True, switch_to_upper=True
+            )
+        # this should never be reached due to SLY rules.
+        # still if it is somehow reached this error is more helpful to the user.
+        except ValueError:  # pragma: no cover
+            raise MalformedInputError(
+                input,
+                f"{self._surface_type.value} could not be parsed as a surface type mnemonic.",
+            )
+        if (
+            self.surface_type is not None
+            and self.surface_type not in self._allowed_surface_types()
+        ):
+            raise ValueError(
+                f"{type(self).__name__} must be a surface of type: {[e.value for e in self._allowed_surface_types()]}"
+            )
+        # parse the parameters
+        for entry in self._tree["data"]:
+            self._surface_constants.append(entry)
+
+    @staticmethod
+    def _number_of_params():
+        """
+        The number of defaults parameters to load into the syntax tree.
+
+        Returns
+        -------
+        int
+        """
+        return 1
 
-    @make_prop_val_node("_surface_type", (SurfaceType, str), SurfaceType)
+    @staticmethod
+    def _allowed_surface_types():
+        """ "
+        The allowed surface types for this surface type.
+
+        Returns
+        -------
+        set[SurfaceType]
+        """
+        return set(SurfaceType)
```

Comment:
Calling `set(arg)` requires `arg` to be iterable. Need:

```python
return {SurfaceType}
```

However, I wonder whether this should just be an abstract method and not implemented at all?